src/target/adiv5_jtag.c

```diff
@@ -57,11 +57,36 @@ void adiv5_jtag_dp_handler(const uint8_t dev_index)
 	bmda_jtag_dp_init(dp);
 #endif
 
-	if (jtag_devs[dev_index].jd_idcode == JTAG_IDCODE_ARM_DPv0)
+	const uint32_t idcode = jtag_devs[dev_index].jd_idcode;
+
+	/* Check if we have a JTAG ID code and that it's a valid one */
+	if (idcode & 1U) {
+		/*
+		 * We do, so this is a JTAG DP. Let's pull out the designer and part codes then.
+		 *
+		 * Start by pulling out the designer code which will be used to attempt to
+		 * detect a DPv0 DP. This will get overriden later by DPIDR if the DP turns out
+		 * to be DPv1+.
+		 */
+		const uint16_t designer = (idcode & JTAG_IDCODE_DESIGNER_MASK) >> JTAG_IDCODE_DESIGNER_OFFSET;
+		/*
+		 * Now extract the part number and sort out the designer code.
+		 * The JTAG ID code designer is in the form:
+		 * Bits 10:7 - JEP-106 Continuation Code
+		 * Bits 6:0 - JEP-106 Identity Code
+		 * So here we convert that into our internal representation.
+		 * See the JEP-106 code list (jep106.h) for more on that.
+		 */
+		dp->designer_code =
+			((designer & ADIV5_DP_DESIGNER_JEP106_CONT_MASK) << 1U) | (designer & ADIV5_DP_DESIGNER_JEP106_CODE_MASK);
+		dp->partno = (idcode & JTAG_IDCODE_PARTNO_MASK) >> JTAG_IDCODE_PARTNO_OFFSET;
+	}
+
+	if (dp->partno == JTAG_IDCODE_PARTNO_DPv0)
```

Comment:
Is not checking for for ARM designer code here intentional? Is the JTAG DP guaranteed to be always ARM in this context?

Reply:
Yes, thanks to the JTAG devices table requiring the ARM manufacturer code to call this function - https://github.com/blackmagic-debug/blackmagic/blob/14f28821bc699bfe9195dd68e13d2abb89aa6fcc/src/target/jtag_devs.c#L28-L35

That means, for now, the ARM part is checked by the scan code prior to entry here - handily!